staking_deposit/utils/validation.py

```diff
@@ -42,16 +43,34 @@
 # Deposit
 #
 
+def _deposit_validator(kwargs: Dict[str, Any]) -> bool:
+    deposit: Dict[str, Any] = kwargs.pop('deposit')
+    credential: Credential = kwargs.pop('credential')
+    return validate_deposit(deposit, credential)
+
+
 def verify_deposit_data_json(filefolder: str, credentials: Sequence[Credential]) -> bool:
     """
     Validate every deposit found in the deposit-data JSON file folder.
     """
+    all_valid_deposits = True
+    deposit_json = []
     with open(filefolder, 'r', encoding='utf-8') as f:
         deposit_json = json.load(f)
-        with click.progressbar(deposit_json, label=load_text(['msg_deposit_verification']),
-                               show_percent=False, show_pos=True) as deposits:
-            return all([validate_deposit(deposit, credential) for deposit, credential in zip(deposits, credentials)])
-    return False
+
+    with click.progressbar(length=len(deposit_json), label=load_text(['msg_deposit_verification']),
+                           show_percent=False, show_pos=True) as bar:
+        executor_kwargs = [{
+            'credential': credential,
+            'deposit': deposit,
```

Comment:
If we change this to `deposit_data_dict` can we then pass in the arguments directly on line 49 instead of poping?

Reply:
I simplified everything by simply reusing the validator_deposit function we already have in b59b0601cd159506106bf0fd60f9e942b3f30144